FileSpecs/CharData.md

```diff
@@ -62,13 +64,94 @@ Offset | Type | Name | Description
+- *Join* will implicitly add the NPC to the party (if possible, otherwise the interaction presumably does not trigger (?)).
+
+
+### Reactions
+
+| Reaction Type | Name          | Arg0   | Arg1 | Arg2      | Meaning                                                                                                            |
+|---------------|---------------|--------|------|-----------|--------------------------------------------------------------------------------------------------------------------|
+| `00`          | None          |        |      |           | No reaction                                                                                                        |
+| `01`          | Say           |        |      | *msg*     | Says string *msg* from the NPC's Dialogue Message table                                                            |
+| `02`          | TeachWord     |        |      | *keyword* | Teaches the party the given *keyword* for future dialogue                                                          |
+| `03`          | GiveItem      |        |      | *index*   | Gives the party a copy of the item at the NPC's item *index* (`0`: first equipped item, `a`: first inventory item) |
+| `04`          | GiveGold      |        |      | *gp*      | Gives the specified number of *gp*                                                                                 |
+| `05`          | GiveFood      |        |      | *amount*  | Gives the specified *amount* of food                                                                               |
+| `06`          | CompleteQuest |        |      | *flag*    | Sets the specified Quest Completion *flag*                                                                         |
+| `07`          | RaiseStat     | *stat* | `1`  | *amount*  | Increases the given *stat* (u16 / Word) by the given *amount*                                                      |
+| `07`          | SetFlag       | *stat* | `4`  | *bitnr*   | Sets the *bitnr*th bit in the given *stat* (u8 / Byte)                                                             |
+
+**Notes**
+- *GiveItem* will clone the NPC's item.  This happens e.g. if the players ask repeatedly for a key to the sewers-- they can get as many keys as they would like.
+- *CompleteQuest* can affect some other NPC's quest completion flag, hence the need for an explicit parameter.  In the game, Mera can set Shir'kar's quest flag, allowing the party to recruit Shir'kar.
+- The NPCs Satine and Shandra share their quest completion flags, so that Satine's interaction options change when Shandra's quest is completed
+- *RaiseStat* uses offsets into character data to identify the stat to raise, so e.g. *stat*=`004E` will raise the "Speed" stat.
+- *RaiseStat* uses offset `00cc` to raise experience points.  This is two bytes before the actual experience point stat; this extra word is likely used to "stage" imminent experience raises
```

Comment:
In Ambermoon the Exp is 4 Bytes long. Maybe it is here as well? Maybe in the case where exp is added via an event the exp is higher than 65535 so it adds a small value to the upper word?

Reply:
That would make some sense-- the instruction would then have to have the 16/32 bit distinction hard-coded, though, unlike for the skill flags, where the NPC interaction code tells it what kind of update it is.
So this needs more checking.